### PR TITLE
Improve debug logging in CLI commands

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: cli/src/main/kotlin/OrtMain.kt [hunk 1]
```diff
@@ -71,7 +71,7 @@ data class GlobalOptions(
     val forceOverwrite: Boolean
 )
 
-class OrtMain : CliktCommand(name = ORT_NAME, invokeWithoutSubcommand = true) {
+object OrtMain : CliktCommand(name = ORT_NAME, invokeWithoutSubcommand = true) {
```
Comment (fviernau) [inline cli/src/main/kotlin/OrtMain.kt:83]: Does this change work well together with the succeeding commit wrt. to our log function (see below)?
I wonder what `this::class.java` evaluates when it's an object. I guess it's an anonymous class but haven't checked.

```
val Any.log: KotlinLogger
    inline get() = loggerOfClass.getOrPut(this::class.java) {
        this::class.qualifiedName?.let { name ->
            require(name.startsWith("org.ossreviewtoolkit.")) {
                "Logging is only allowed on ORT classes, but '$name' is used."
            }
        }

        loggerOf(this::class.java)
    }
```

File: cli/src/main/kotlin/OrtMain.kt [hunk 2]
```diff
@@ -210,6 +215,20 @@ object OrtMain : CliktCommand(name = ORT_NAME, invokeWithoutSubcommand = true) {
     }
 }
 
+/**
+ * Write the [ortResult] to all [outputFiles] for the given [resultName].
+ */
+fun CliktCommand.writeOrtResult(ortResult: OrtResult, outputFiles: Collection<File>, resultName: String) {
+    outputFiles.forEach { file ->
+        println("Writing $resultName result to '$file'.")
```
Comment (fviernau) [inline cli/src/main/kotlin/OrtMain.kt:241]: Have you considered moving the `println()` above the for loop, like:
`println("Writing $resultName result to '$outputFiles'.")

The output IMO gives kind of a heads up that this can take a little bit.
So that heads up can be given right away for all files which are planned to be written.

File: cli/src/main/kotlin/OrtMain.kt [hunk 3]
```diff
@@ -45,6 +46,7 @@ import org.ossreviewtoolkit.commands.*
 import org.ossreviewtoolkit.model.OrtResult
 import org.ossreviewtoolkit.model.config.LicenseFilenamePatterns
 import org.ossreviewtoolkit.model.config.OrtConfiguration
+import org.ossreviewtoolkit.model.readValue
```
Comment (fviernau) [inline cli/src/main/kotlin/OrtMain.kt:50]: Consider swapping the order of this commit with the previous one.

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt [hunk 4]
```diff
@@ -229,39 +228,33 @@ class EvaluatorCommand : CliktCommand(name = "evaluate", help = "Evaluate ORT re
             }
         }
 
-        var (ortResultInput, readDuration) = measureTimedValue { ortFile?.readValue<OrtResult>() }
-
-        ortFile?.let { file ->
-            log.perf {
-                "Read ORT result from '${file.name}' (${file.formatSizeInMib}) in ${readDuration.inMilliseconds}ms."
-            }
+        val existingOrtFile = requireNotNull(ortFile) {
```
Comment (fviernau) [inline cli/src/main/kotlin/commands/EvaluatorCommand.kt:231]: Moving this here could be a separate commit preceding this one.
IMO it would be nice to at least mention that: while at it this check has been moved further up to fail early.

**Review by fviernau — APPROVED** [review]

File: cli/src/main/kotlin/OrtMain.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/OrtMain.kt:241]: No, I haven't considered this (as it was like that before). But what's the advantage of giving this "heads-up" for all files at once? Isn't it even better the way it is now, where the heads-up is given before each file is written, and then by the time of the next heads-up you know that the previous file was fully written? Like a crude progress indicator?

File: cli/src/main/kotlin/OrtMain.kt (hunk 3 above)
Comment (sschuberth) [reply cli/src/main/kotlin/OrtMain.kt:50]: I liked it better to first refactor the advisor / scanner to have the extracted code at the same location as the other subcommands, and then extract the code.

File: cli/src/main/kotlin/commands/EvaluatorCommand.kt (hunk 4 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/EvaluatorCommand.kt:231]: > Moving this here could be a separate commit preceding this one.

You're right, done.

**Review by fviernau — APPROVED** [review]
